src/main/java/rest/InvoiceResource.kt

```diff
@@ -86,9 +86,8 @@ class InvoiceResource @Inject constructor(
     @Path("/regenerate/{uuid}")
     @Produces("application/json")
     fun regenerateInvoice(@PathParam("uuid") uuid: String): Response {
-        val regeneratedInvoice = invoiceService.regenerateInvoice(uuid)
-            ?: return Response.notModified().build()
+        invoiceService.regenerateInvoiceMQ(uuid)
 
-        return Response.ok(regeneratedInvoice).build()
+        return Response.ok("Invoice is being regenerated").build()
```

Comment:
Maybe return a reference to the invoice? tht way we can keep checking it on the frontend